### PR TITLE
feat(ui): Add Channel Header customization

### PR DESCRIPTION
**Pull request by S-ecki**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request

- added properties `elevation` and `centerTitle` on `ChannelHeader`
- previous values are taken as defaults if properties are null (so change does not affect existing designs)
- this was done to enable users to style the `ChannelHeader` like an `AppBar` (e.g. elevation = 4 and centerTitle = false by default)

### CODE REVIEW

**Review by imtoori — CHANGES_REQUESTED** [review]
hey @S-ecki thanks for the PR
I polished it a bit

I think `centerTitle` does not do anything right now, can you confirm this? Maybe we can remove it or find a way of making it effective.

Also, I think we should do the same for `ThreadHeader` and `ChannelListHeader`
What do you think?

**Comment by S-ecki** [comment]
hey @imtoori, thanks for the answer!

`centerTitle` does indeed change something for me. I looked further into the issue and discovered that the Flutter `AppBar` sets the default for `centerTitle` based on the Platform. You can find the logic in Flutter´s `app_bar.dart` file on line 714 (`_getEffectiveCenterTitle()`).
TLDR: on Android, `centerTitle` is false. on iOS it is true unless there are 2 or more `actions` in the `AppBar`.

I would try to align `ChannelHeader`s `centerTitle` with Flutter `AppBar` `centerTitle`. 

The reason why I want it to be consistent is following: I have a mobile App with multiple Tabs, where each Tab has a AppBar with the same design. One of those Tabs contains Stream Chat and I want the ChannelHeader to be used as the AppBar. Thus, it needs to be styled exactly the same.

I did not use `ThreadHeader` or `ChannelListHeader` yet, but I would also keep them consistent with `AppBar` 👍

**Comment by S-ecki** [comment]
hey @imtoori!

I tested around with the `centerTitle` property and came to the conclusion that the best way to handle it is to let the actual `AppBar` handle it. What I mean by that is a user can pass in a `centerTitle` property, which will be set. If no `centerTitle` is passed in, we pass `null` to the Flutter `AppBar`. By that, the value of `centerTitle` gets decided based on `_getEffectiveCenterTitle()` (like mentioned in my last comment).
This way, we can ensure that Stream Chat centerTitle properties are always aligned with Flutter AppBar.

I also added elevation and centerTitle to `ThreadHeader` and `ChannelListHeader`.

**Comment by S-ecki** [comment]
Hey @imtoori 

are there any updates on this issue? Any improvements I could make?

**Comment by imtoori** [comment]
hey @S-ecki I'm reviewing it now, sorry about the delay but I was focused on another task

**Comment by imtoori** [comment]
@S-ecki the only problem I see is that I don't see any visual change after changing the `centerTitle` property to true/false
I think that's because we don't use just a `Text`, but a more complex widget that messes up with that property
The title stays always centered using the default configuration changing only the `centerTitle` property
Can you confirm this? I'm pretty sure we can implement a workaround for this

**Comment by S-ecki** [comment]
Hey @imtoori, I unfortunately can confirm that the `centerTitle` property does not change anything visually, sorry!
Do you have an idea of how we can work around that?

**Comment by imtoori** [comment]
I'm working on #995 at the moment and it's quite high in priority, I'll see what I can do later today hopefully

**Comment by S-ecki** [comment]
sorry to bother you again @imtoori , any updates in sight? :)

**Comment by imtoori** [comment]
hey @S-ecki not yet, I'm sorry, I'll work on this in the following days

**Comment by peter-gy** [comment]
Hey @imtoori, I also find myself in need to have this issue resolved. If you have a vague idea about the approach to be taken to solve the problem, I would be glad to contribute!

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 1]
```diff
@@ -61,9 +61,11 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
     this.showConnectionStateTile = false,
     this.title,
     this.subtitle,
+    this.centerTitle,
     this.leading,
     this.actions,
     this.backgroundColor,
+    this.elevation,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:68]:
```dart
this.elevation = 1
```

Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/channel_header.dart:68]:
```suggestion
    this.elevation = 1,
```

File: packages/stream_chat_flutter/lib/src/thread_header.dart [hunk 2]
```diff
@@ -179,4 +191,21 @@ class ThreadHeader extends StatelessWidget implements PreferredSizeWidget {
 
   @override
   final Size preferredSize;
+
+  bool _getEffectiveCenterTitle(ThemeData theme) {
```
Comment (ayush221b) [inline packages/stream_chat_flutter/lib/src/thread_header.dart:195]: Since we are using the same function in `ChannelHeader` as well as `ThreadHeader` do you think it would be a good idea to make it a global helper function and use that in both places, instead of duplicating the code?

**Comment by S-ecki** [comment]
thanks a lot for merging @imtoori! Is there a timeline for release 3.6.2?

**Comment by imtoori** [comment]
We're currently focused on releasing v4 today, 3.7 will come after that
